File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargeImageListFragment.java
package com.didichuxing.doraemonkit.kit.largepicture;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.kit.blockmonitor.BlockListAdapter;
import com.didichuxing.doraemonkit.kit.blockmonitor.bean.BlockInfo;
import com.didichuxing.doraemonkit.kit.blockmonitor.core.BlockMonitorManager;
import com.didichuxing.doraemonkit.ui.base.BaseFragment;
import com.didichuxing.doraemonkit.ui.widget.recyclerview.DividerItemDecoration;
import com.didichuxing.doraemonkit.ui.widget.titlebar.TitleBar;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

/**
 * @desc: 大图检测列表
 */

public class LargeImageListFragment extends BaseFragment {
    private static final String TAG = "LargeImageListFragment";

    private RecyclerView mLargeImageList;
    private LargeImageListAdapter mLargeImageListAdapter;
    private TitleBar mTitleBar;

    @Override
    protected int onRequestLayout() {
        return R.layout.dk_fragment_large_img_list;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
        load();
    }


    private void initView() {
        mLargeImageList = findViewById(R.id.block_list);

        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        mLargeImageList.setLayoutManager(layoutManager);
        mLargeImageListAdapter = new LargeImageListAdapter(getContext());
        mLargeImageList.setAdapter(mLargeImageListAdapter);
        DividerItemDecoration decoration = new DividerItemDecoration(DividerItemDecoration.VERTICAL);
        decoration.setDrawable(getResources().getDrawable(R.drawable.dk_divider));
        mLargeImageList.addItemDecoration(decoration);

        mTitleBar = findViewById(R.id.title_bar);
        mTitleBar.setOnTitleBarClickListener(new TitleBar.OnTitleBarClickListener() {
            @Override
            public void onLeftClick() {
                getActivity().onBackPressed();
            }

            @Override
            public void onRightClick() {

            }
        });
    }


    private void load() {
        List<LargeImageInfo> infos = new ArrayList<>();
        for (LargeImageInfo largeImageInfo : LargePictureManager.LARGE_IMAGE_INFO_MAP.values()) {
            infos.add(largeImageInfo);
        }
        mLargeImageListAdapter.setData(infos);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/largepicture/LargeImageListFragment.kt
package com.didichuxing.doraemonkit.kit.largepicture

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.config.PerformanceSpInfoConfig.getLargeImgFileThreshold
import com.didichuxing.doraemonkit.config.PerformanceSpInfoConfig.getLargeImgMemoryThreshold
import com.didichuxing.doraemonkit.kit.core.BaseFragment
import com.didichuxing.doraemonkit.widget.titlebar.TitleBar
import java.util.*

/**
 * @author:  maple
 * @time:  2020/6/10 - 16:22
 * @desc: 大图检测列表
 */
class LargeImageListFragment : BaseFragment() {
    private lateinit var mLargeImageList: RecyclerView
    private lateinit var mLargeImageListAdapter: LargeImageListAdapter
    private lateinit var mTitleBar: TitleBar
    override fun onRequestLayout(): Int {
       return R.layout.dk_fragment_large_img_list
    }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        load()
    }
    private fun initView() {
        mLargeImageList = findViewById(R.id.block_list)
        val layoutManager = LinearLayoutManager(context)
        mLargeImageList.layoutManager = layoutManager
        mLargeImageListAdapter = LargeImageListAdapter()
        mLargeImageList.adapter = mLargeImageListAdapter
        val decoration = DividerItemDecoration(context,DividerItemDecoration.VERTICAL)
        decoration.setDrawable(resources.getDrawable(R.drawable.dk_divider))
        mLargeImageList.addItemDecoration(decoration)
        mTitleBar = findViewById(R.id.title_bar)
        mTitleBar.onTitleBarClickListener = object : TitleBar.OnTitleBarClickListener {
            override fun onLeftClick() {
                activity!!.onBackPressed()
            }

            override fun onRightClick() {}
        }
    }
    private val fileThreshold = getLargeImgFileThreshold(LargePictureManager.FILE_DEFAULT_THRESHOLD)
    private val memoryThreshold = getLargeImgMemoryThreshold(LargePictureManager.MEMORY_DEFAULT_THRESHOLD)


    private fun load() {
        val imageInfos: MutableList<LargeImageInfo> = ArrayList()
        for (largeImageInfo in LargePictureManager.LARGE_IMAGE_INFO_MAP.values) {
            if (largeImageInfo.getFileSize() < fileThreshold && largeImageInfo.getMemorySize() < memoryThreshold) {
                continue
            }
            imageInfos.add(largeImageInfo)
        }
        mLargeImageListAdapter.setList(imageInfos)
    }
}